core/blockstm/dag.go

```diff
@@ -69,6 +75,58 @@ func BuildDAG(deps TxnInputOutput) (d DAG) {
 	return
 }
 
+func UpdateDeps(deps map[int]map[int]bool, t DepsChan) map[int]map[int]bool {
+	txTo := t.ReadList
+
+	deps[t.Index] = map[int]bool{}
+
+	for j := 0; j <= t.Index-1; j++ {
+		txFrom := t.FullWriteList[j]
+
+		if HasReadDep(txFrom, txTo) {
+			deps[t.Index][j] = true
+
+			for k := range deps[t.Index] {
+				_, boool1 := deps[j][k]
+
+				if boool1 {
+					delete(deps[t.Index], k)
+				}
+			}
+		}
+	}
+
+	return deps
+}
+
+func GetDep(deps TxnInputOutput) map[int]map[int]bool {
+	newDependencies := map[int]map[int]bool{}
+
+	for i := 1; i < len(deps.inputs); i++ {
+		txTo := deps.inputs[i]
+
+		newDependencies[i] = map[int]bool{}
+
+		for j := 0; j <= i-1; j++ {
+			txFrom := deps.allOutputs[j]
+
+			if HasReadDep(txFrom, txTo) {
+				newDependencies[i][j] = true
+
+				for k := range newDependencies[i] {
+					_, boool1 := newDependencies[j][k]
+
+					if boool1 {
+						delete(newDependencies[i], k)
+					}
+				}
+			}
+		}
```

Comment:
So I also tried doing something like this:
```
func depsHelper(dependencies map[int]map[int]bool, allOutputs interface{}, txTo TxnInput, i int, interfaceFlag int) map[int]map[int]bool {
	var allOutputs1 interface{}

	if interfaceFlag == 1 {
		allOutputs1 = allOutputs.([][]WriteDescriptor)
	} else if interfaceFlag == 2 {
		allOutputs1 = allOutputs.([]TxnOutput)
	}

	for j := 0; j <= i-1; j++ {
		txFrom := allOutputs1[j]

		if HasReadDep(txFrom, txTo) {
			dependencies[i][j] = true

			for k := range dependencies[i] {
				_, foundDep := dependencies[j][k]

				if foundDep {
					delete(dependencies[i], k)
				}
			}
		}
	}

	return dependencies
}

func UpdateDeps(deps map[int]map[int]bool, t TxDep) map[int]map[int]bool {
	txTo := t.ReadList

	deps[t.Index] = map[int]bool{}

	deps = depsHelper(deps, t.FullWriteList, txTo, t.Index, 1)

	return deps
}

func GetDep(deps TxnInputOutput) map[int]map[int]bool {
	newDependencies := map[int]map[int]bool{}

	for i := 1; i < len(deps.inputs); i++ {
		txTo := deps.inputs[i]

		newDependencies[i] = map[int]bool{}

		newDependencies = depsHelper(newDependencies, deps.allOutputs, txTo, i, 2)
	}

	return newDependencies
}

```

But because of the different type of `deps.allOutputs` and `t.Index`, I cannot do that.